tests/core_test/containers_test/mels_repertoire_test.py

```diff
@@ -0,0 +1,230 @@
+    fake_extra_scores: ExtraScores = {}
+
+    # do an addition
+    repertoire = repertoire.add(
+        fake_genotypes, fake_descriptors, fake_fitnesses, fake_extra_scores
+    )
+
+    # Either solution may be added due to the behavior of jax.at[].set():
+    # https://jax.readthedocs.io/en/latest/_autosummary/jax.numpy.ndarray.at.html
+    # Thus, we provide possible values for each scenario.
+
+    # check that the repertoire looks like expected
+    expected_genotypes_1 = jnp.array(
+        [
+            [2.0 for _ in range(genotype_size)],
+            [0.0 for _ in range(genotype_size)],
+            [0.0 for _ in range(genotype_size)],
+            [0.0 for _ in range(genotype_size)],
+        ]
+    )
+    expected_fitnesses_1 = jnp.array([1.0, -jnp.inf, -jnp.inf, -jnp.inf])
+    expected_descriptors_1 = jnp.array(
+        [
+            [1.0, 1.0],  # Centroid coordinates.
+            [0.0, 0.0],
+            [0.0, 0.0],
+            [0.0, 0.0],
+        ]
+    )
+    expected_spreads_1 = jnp.array([0.75, jnp.inf, jnp.inf, jnp.inf])
+
+    expected_genotypes_2 = jnp.array(
+        [
+            [3.0 for _ in range(genotype_size)],
+            [0.0 for _ in range(genotype_size)],
+            [0.0 for _ in range(genotype_size)],
+            [0.0 for _ in range(genotype_size)],
+        ]
+    )
+    expected_fitnesses_2 = jnp.array([0.5, -jnp.inf, -jnp.inf, -jnp.inf])
+    expected_descriptors_2 = jnp.array(
+        [
+            [1.0, 1.0],  # Centroid coordinates.
+            [0.0, 0.0],
+            [0.0, 0.0],
+            [0.0, 0.0],
+        ]
+    )
+    expected_spreads_2 = jnp.array([0.5, jnp.inf, jnp.inf, jnp.inf])
+
+    # check values
+    pytest.assume(
+        jnp.allclose(repertoire.genotypes, expected_genotypes_1, atol=1e-6)
+        or jnp.allclose(repertoire.genotypes, expected_genotypes_2, atol=1e-6)
+    )
+
+    if jnp.allclose(repertoire.genotypes, expected_genotypes_1, atol=1e-6):
+        pytest.assume(
+            jnp.allclose(repertoire.genotypes, expected_genotypes_1, atol=1e-6)
+        )
+        pytest.assume(
+            jnp.allclose(repertoire.fitnesses, expected_fitnesses_1, atol=1e-6)
+        )
+        pytest.assume(
+            jnp.allclose(repertoire.descriptors, expected_descriptors_1, atol=1e-6)
+        )
+        pytest.assume(jnp.allclose(repertoire.spreads, expected_spreads_1, atol=1e-6))
+    elif jnp.allclose(repertoire.genotypes, expected_genotypes_2, atol=1e-6):
+        pytest.assume(
+            jnp.allclose(repertoire.genotypes, expected_genotypes_2, atol=1e-6)
+        )
+        pytest.assume(
+            jnp.allclose(repertoire.fitnesses, expected_fitnesses_2, atol=1e-6)
+        )
+        pytest.assume(
+            jnp.allclose(repertoire.descriptors, expected_descriptors_2, atol=1e-6)
+        )
+        pytest.assume(jnp.allclose(repertoire.spreads, expected_spreads_2, atol=1e-6))
+
+
+def test_single_eval() -> None:
```

Comment:
Minor suggestion: we could use a short sentence to define the scope of the test

Reply:
Updated the test names and added docstrings